Added scrolling support to Mouse action.

wolfmanstout opened the pull request.

sathishkottravel committed:
Dragonfly: Python35 prorting.
1) 2to3 has been used initially
2) string.letter -> string.ascii_letters
3) c_char(char)  -> c_char(ord(char)) , similar c_wchar
4) __str__(self) modified in action_base.py and state.py  ( #http://stackoverflow.com/questions/6812031/how-to-make-unicode-string-with-python3)

sathishkottravel committed:
Dragonfly: Python35 prorting. temp fix.

wolfmanstout committed:
Added scrolling support to Mouse action.

drmfinlay committed:
Ignored IDE files

drmfinlay committed:
Somewhat implemented the dragonfly backend for CMU Sphinx/Pocket Sphinx. There\'s still lots to do

drmfinlay committed:
Added check for Sphinx engine to engines/__init__

Commit:
Add support for different keyboard layouts

drmfinlay committed:
Placeholder import of defunct (at least for me) pocketsphinx Python package for connection / availability of pocket sphinx engine

drmfinlay committed:
Fixed PEP8 issues in state.py

drmfinlay committed:
Wrote a JSGF Python module + some tests for interaction with Sphinx's JSGF grammar options

drmfinlay committed:
Wrote and tested dragonfly2jsgf.py to translate dragonfly grammars and their components into the equivalent JSGF components from the JSGF module

drmfinlay committed:
Fix some style and Python type + name redeclaration issues in 2 files

drmfinlay committed:
Specify more types in docstrings

drmfinlay committed:
Add end of string match '\Z' in Rule.matches() after expansion.matching_regex() so that there aren't false matches

drmfinlay committed:
Change __init__.py files for dragonfly modules to import new mocked classes for Windows-specific parts of dragonfly when the system platform is not Windows

drmfinlay committed:
Fix a bug with some JSGF Expansion classes where the wrong value is passed to the super constructor

drmfinlay committed:
Override __str__() for RuleRef to display 'RuleRef('rule_name')'

drmfinlay committed:
Add GrammarError class in jsgf/grammar for JSGF grammar errors

drmfinlay committed:
Add dependencies property to Rule class for finding all dependenies for a Rule and any dependent rule

drmfinlay committed:
Add new unit test file and test case to test dependencies property of Rule class

drmfinlay committed:
Make Windows-dependent calls in dragonfly/log.py conditional on whether the win32com.shell imports succeed or if the platform is Windows

drmfinlay committed:
Fix various code style and semantic issues in dragonfly/log.py

drmfinlay committed:
Import mocked Windows-only classes in grammar_connection.py if the platform is non-Windows

drmfinlay committed:
Fix code style and semantic issues in grammar_connection.py

drmfinlay committed:
Lower Literal text so that Pocket Sphinx recognizes words properly in the dictionary

drmfinlay committed:
Fix some expected values in dragonfly2jsgf_test cases by changing words to lowercase

drmfinlay committed:
Replace overcomplicated code for handling special whitespace cases in speech text with a simpler approach

drmfinlay committed:
Add Dictation Expansion subclass for representing dictation input in JSGF rules

drmfinlay committed:
Remove unintended line from docstring

drmfinlay committed:
Change 'Import' to __class__.__name__ in __str__ method of grammar Import class

drmfinlay committed:
Add docstrings for 4 methods in JSGF Grammar class

drmfinlay committed:
Add visible_rules and rule_names properties to JSGF Grammar class

drmfinlay committed:
Add remove_rule method to JSGF Grammar class

drmfinlay committed:
Change find_matching_rules method of JSGF Grammar class to only match visible rules so that internal rules can't be matched instead

drmfinlay committed:
Add and test reference_count property in JSGF Rule class and add logic to increment/decrement on creation/deletion of RuleRef objects

drmfinlay committed:
Change name of a grammar_test case and test method

drmfinlay committed:
Add grammar test case for the visible_rules property

drmfinlay committed:
Change docstring of dependencies property of the JSGF Rule class

drmfinlay committed:
Prevent cycles in JSGF Rule dependencies property

drmfinlay committed:
Fix problems with JSGF Grammar remove_rule method

drmfinlay committed:
Move rule and grammar creating code in BasicGrammarCase to setUp method and write a test method for removing dependent rules

drmfinlay committed:
Rewrote VisibleRulesCase test class using setUp method

drmfinlay committed:
Add RootGrammar class for compiling and matching against a list of rules referenced in an AlternativeSet expansion as the only visible rule

drmfinlay committed:
Change compile_with_root_public_rule Grammar method to recreate_with_root_public_rule which returns a new RootGrammar object

drmfinlay committed:
Add test case class testing RootGrammar

drmfinlay committed:
Add parent property to JSGF Expansion base class

drmfinlay committed:
Set the parent property to self for the children of expansions

drmfinlay committed:
Add expansion_test.py and test the parent property in a new TestCase class

drmfinlay committed:
Extract dragonfly2jsgf and unit test from jsgf module into a new one

drmfinlay committed:
Sort the import statements in jsgf __init__.py

drmfinlay committed:
Remove Dictation expansion class from jsgf module

drmfinlay committed:
Add Dictation JSGF expansion class to dragonfly2jsgf module in a new file

drmfinlay committed:
Fix import issues and a reference issue with RuleRef expansion in RootGrammar remove_rule method

drmfinlay committed:
Rename unit test file for dragonfly2jsgf

drmfinlay committed:
Add pyjsgf and sphinxwrapper as dependencies

drmfinlay committed:
Remove jsgf package

drmfinlay committed:
Organise imports and add a few more in dragonfly2jsgf

drmfinlay committed:
Change TranslationState class expansion member to a property

drmfinlay committed:
Add and use JSGFImpossible class in dragonfly2jsgf

drmfinlay committed:
Minor changes in dragonfly2jsgf package

drmfinlay committed:
Remove duplicate declaration of set_exclusiveness method

drmfinlay committed:
Fix style issues in grammar_base.py

drmfinlay committed:
Fix 3 issues in dragonfly2jsgf

drmfinlay committed:
Implement SphinxJSGFCompiler for Sphinx engine using dragonfly2jsgf

drmfinlay committed:
Remove unnecessary overwritten methods from SphinxDictationContainer

drmfinlay committed:
Use sphinxwrapper package in is_engine_available function instead of pocketsphinx

drmfinlay committed:
Change dragonfly2jsgf translate_rule method to use TranslationState instead

drmfinlay committed:
Write and test dragonfly2jsgf translation of dragonfly lists

drmfinlay committed:
Docstring changes in dragonfly2jsgf

drmfinlay committed:
Write and test dragonfly DictList translation in dragonfly2jsgf

drmfinlay committed:
Docstring changes in backend_sphinx

drmfinlay committed:
Update speak and _get_language methods for the Sphinx engine

drmfinlay committed:
Add rule_names property in base Grammar class

drmfinlay committed:
Add missing member declaration in base engine class

wolfmanstout commented:
@t4ngo friendly ping. This is a generic addition to Dragonfly and I've tried to follow your style closely.

drmfinlay committed:
Fix import issues in test_dragonfly2jsgf.py

drmfinlay committed:
Fix dragonfly2jsgf translate_rule_ref method and add a test for it

drmfinlay committed:
Fix typo in elements_basic.py

drmfinlay committed:
Fix error in Context.copy method

drmfinlay committed:
Fix some style issues in context.py

drmfinlay committed:
Add pyaudio to setup requirements

drmfinlay committed:
Remove unnecessary code from Sphinx engine's recobs.py file

drmfinlay committed:
Implement engine and add default engine config module

drmfinlay committed:
Add unit test file for Sphinx engine

drmfinlay committed:
Add Sphinx engine test file to test/suites.py file

drmfinlay committed:
Add example dragonfly module for the Sphinx engine

drmfinlay committed:
Make example Sphinx dragonfly module runnable without a loader

drmfinlay committed:
Update .gitignore

drmfinlay committed:
Add README file for Sphinx engine

drmfinlay committed:
Add section on CMU Sphinx engine

reckoner commented:
@wolfmanstout  I like this addition, but it would help to have some documentation how to use it. For example:

   Mouse('scrollup:10').execute()

or something more explicit?

drmfinlay committed:
Change Windows platform conditionals to use startswith because macOS darwin is not Windows

drmfinlay committed:
Fix NEXT_PART_TIMEOUT config breaking tests sometimes

drmfinlay committed:
Change comment in test file

drmfinlay committed:
Fix issues preventing Sphinx engine and example running on Windows

drmfinlay committed:
Change _last_recognition_time to be set on recognition of rule parts + adjust tests

drmfinlay committed:
Add link and section on improving speech recognition accuracy

drmfinlay committed:
Simplify loop in Sphinx engine method

drmfinlay committed:
Change next part timeout to use seconds instead of milliseconds

drmfinlay committed:
Move Repetition code to before Sequence in dragonfly2jsgf get_jsgf_equiv method

drmfinlay committed:
Change JSGF rule visibility to depend on exported value of dfly rules + adjust tests

drmfinlay committed:
Add 2 read-only properties to Repetition element class

wolfmanstout committed:
Added documentation to mouse scrolling actions.

drmfinlay committed:
Uncomment some methods in RecognitionObserver base class

drmfinlay committed:
Simplify dfly word list generation and related method in Sphinx engine

drmfinlay committed:
Add and test new next_rule_part observer functionality

drmfinlay committed:
Use on_next_rule_part observer method in Sphinx engine example

drmfinlay committed:
Add and test dragonfly2jsgf translation of repeated Dictation elements

drmfinlay committed:
Add and test some engine config validation

drmfinlay committed:
Change setup.py to only require pywin32 on Windows platforms

drmfinlay committed:
Separate Sphinx engine tests into multiple test classes

drmfinlay committed:
Add Gitter chat badge to README

drmfinlay committed:
Fix word case issues in dragonfly Literal translation

drmfinlay committed:
Rename Sphinx engine example module

drmfinlay committed:
Fix Sphinx + dragonfly word case issues during Literal decoding instead

drmfinlay committed:
Write Sphinx engine module loader based on dfly-loader-wsr.py

drmfinlay committed:
Add module loader info to README

drmfinlay committed:
Add testing and examples section to engine README

drmfinlay committed:
Fix NoneType bug in Literal.decode method

drmfinlay committed:
Update recognition observers in Sphinx example and loader

drmfinlay committed:
Change recognition time-out log message

drmfinlay committed:
Update Sphinx engine example

drmfinlay committed:
Set up the Sphinx engine logger in init

drmfinlay committed:
Fix minor issues in engine tests + simplify testing method

drmfinlay committed:
Ignore dependent rules on unloading grammars

drmfinlay committed:
Use new RootDictationGrammar class in Sphinx engine (pyjsgf changes)

drmfinlay committed:
Move Sphinx engine GrammarWrapper class to new file

drmfinlay committed:
Add grammar load checks in Sphinx engine tests

drmfinlay committed:
Fix minor issue with test functions in Sphinx engine tests

drmfinlay committed:
Add 2 test methods to Sphinx engine test class

drmfinlay committed:
Expose engine's observer manager and logger through properties

drmfinlay committed:
Add pyjsgf and sphinxwrapper dependencies as git submodules

drmfinlay committed:
Update README with submodule install instructions

drmfinlay committed:
Do a lot of Sphinx engine redesign...

Including the following:
- use one Pocket Sphinx JSGF search (at least) per grammar
- reprocess recorded speech using each grammar's PS search
- move lots of work to the GrammarWrapper class
- use a state class to collect and process rules instead a list
- get the best hypothesis if multiple searches match recorded speech
- use a Timer thread and lock for next rule part timeout

drmfinlay committed:
Update engine to-do list

drmfinlay committed:
Minor changes to Sphinx test and engines package

drmfinlay committed:
Add Sphinx engine test method

drmfinlay committed:
Add active_rules property to Grammar class + fix typo

drmfinlay committed:
Add context checking - mostly in _speech_start_callback

drmfinlay committed:
Do context checking for the best state in _on_next_part_timeout

drmfinlay committed:
Change _on_next_part_timeout to calculate the best state to use

instead of using a parameter because the given state might be invalid from a context change

wolfmanstout committed:
Support wheeldown/wheelup as alternatives to scrolldown/scrollup in Mouse action to align with Aenea.

drmfinlay committed:
Change a few mocked objects, classes and functions

Most notably, making an instance of an action class (Text, Key, etc)
will return either an ActionBase or DynStrActionBase object and
accept variable arguments, so they will not break in MappingRules
on non-Windows systems anymore. This is also useful for using aenea.

drmfinlay committed:
Update dragonfly2jsgf & setup.py for pyjsgf v1.0.1

Note that v1.0.1 includes moving from distutils to setuptools.
This means that previous versions of pyjsgf should be uninstalled
or removed from the dist-packages folder where Python is installed.

drmfinlay committed:
Update submodule install instructions + update pyjsgf submodule

drmfinlay committed:
Update README and setup.py to make Sphinx engine optional

drmfinlay committed:
Update imports in two __init__.py files

drmfinlay committed:
Change Repetition translation + clean up dragonfly2jsgf tests

drmfinlay committed:
Update pyjsgf submodule

drmfinlay committed:
Add IntRef test to dragonfly2jsgf tests

drmfinlay committed:
Update pyjsgf submodule + required version to v1.2.0

drmfinlay committed:
Fix styling + make natlink optional in dragonfly/timer.py

drmfinlay committed:
Make monitor list and rectangles update (fix issue #7)

- Monitor list is updated with Monitor.update_monitors_list()
- Rectangles and Points are also now comparable
- Fix some style issues in monitor.py

drmfinlay committed:
Use PatchedRepeat class as a workaround for PS bug

This just compiles Repetition as "<element> [<element>]*" instead
of "<element>+" to avoid a CMU Sphinx bug with the JSGF repeat (+)
operator.

drmfinlay committed:
Enable threading support for natlink engine

natConnect(True) should stop Windows COM callbacks from blocking
Python interpreter threads

drmfinlay committed:
Merge branch 'master' of github.com:sathishkottravel/dragonfly into feat/py3-support

I've reverted the Unicode changes in favour of using 2to3 from
setuptools to translate for Python 3 with only a few fixes
enabled. The same has been done for basestring, str and long->int
changes. There are also still a few newer issues to be resolved
with 2to3.

There are also a few unrelated changes in apps/family and some
typo fixes.

drmfinlay committed:
Make dragonfly compatible with Python 3 and above

- 2to3 was mostly used to unify syntactic differences between
  versions. E.g. 'except Exception, e' -> 'except Exception as e'
  and 'print "text"' -> 'print("text")'. It was also used for
  generator function changes and relative imports.
- The 'six' package has been used for semantic differences between
  versions. E.g. unicode -> str conversion. That package has been
  added to install_requires in setup.py.
- Not sure if I've done the windows-1252 encoding correctly across
  Python versions

drmfinlay committed:
Fix bug in two Keyboard class methods

drmfinlay committed:
Make SphinxEngine.mimic method work with Mimic action + add tests

- Most rules/mappings can be mimicked now with one Mimic
  action/call.
- Old engine.mimic method was renamed to mimic_phrases and the
  tests were adjusted.
- The new test currently fails at the end due to a pyjsgf
  limitation with matching ambiguous expansions.
- Update pyjsgf required version and submodule to v1.2.2

drmfinlay committed:
Fix a few issues in dragonfly/parser and its tests

- Fix UnicodeDecodeError that occurred when unencoded Unicode
  strings were used in the String or CharacterSeries parser
  classes (issue #12)
- Make doctests pass by removing asserted isinstance calls
- Test Unicode strings in parser tests
- Fix some formatting and style issues in test file

drmfinlay committed:
Rewrite Clipboard class using pyperclip + add tests

See issue #15.
- This makes the clipboard functionality work on other platforms.
- The only supported Windows clipboard format is Unicode text.
- The class should be backwards-compatibile assuming Unicode text
  is the only clipboard format used.

drmfinlay committed:
Clean up & adjust action_paste.py for Clipboard changes

comodoro committed:
Fix Paste action for Unicode

drmfinlay committed:
Properly set the language for Sphinx JSGF grammars

drmfinlay committed:
Require sphinxwrapper v1.1.1 + minor changes

- Update sphinxwrapper submodule and setup.py version
- Make some minor changes in Sphinx engine and its tests

drmfinlay committed:
Add additional Sphinx engine functionality and tests

- Add wake up/sleep functionality with options in the default
  config module.
- Add start asleep config option (default is on).
- Add set_keyphrase & unset_keyphrase engine methods for adding
  additional key phrases (issue #3).
- Extract some code from recognise_forever method into a new
  process_buffer method that can be used for streaming audio from
  other sources (issue #6).

drmfinlay committed:
Merge branch 'feat/py3-support' into develop

Closes #10.

- Not everything is working yet for Python 3.
- There may be still be some issues with Python 2 (a lot of code
  changes), so this is being merged into 'develop' first. Any
  issues will probably be related to encoding between
  windows-1252, utf-8 and/or ascii.

drmfinlay committed:
Fix issues in dragonfly/language/number_arabic.py

drmfinlay committed:
Add missing return in Keyboard method + remove unused import

drmfinlay committed:
Allow non-ascii (Unicode) characters for dragonfly specs

Closes issue #14.
This works with the natlink engine and can work with the sphinx
engine when using Python 3.x if the words are in the active
pronunciation dictionary.

drmfinlay committed:
Fix a few minor problems in 3 files

drmfinlay committed:
Split Clipboard class into separate classes

The Windows-only Clipboard class has been left for backwards-
compatibility. The new multi-platform class has been moved to
dragonfly.util.clipboard, with a few changes.

Both classes inherit from a BaseClipboard class.

drmfinlay committed:
Merge branch 'feat/multi-platform-clipboard' into develop

Closes #15.

drmfinlay committed:
Add non-ascii clipboard test and pyperclip bug workaround

Pyperclip version 1.6.1 is required until pyperclip issue 129 is
is resolved.

drmfinlay committed:
Update natlink links in documentation

drmfinlay committed:
Update to version 0.7.0

drmfinlay committed:
Release dragonfly version 0.7.0

Major changes are:
- Python 3.x support
- Multi-platform clipboard class
- Unicode support for dragonfly specs
- Bug fixes

drmfinlay committed:
Add CHANGELOG.md and remove empty CHANGES.txt file

The new file uses the "Keep a Changelog" format. I'm removing the
CHANGES.txt file to avoid confusion.

drmfinlay committed:
Add initial Travis CI .yml file

This should run the unit and doc tests for dragonfly's parser.

drmfinlay committed:
Fix syntax error in .travis.yml

drmfinlay committed:
Add Travis CI build status badge to README

drmfinlay committed:
Update CHANGELOG.md

Add new headings and references for older release versions.

wolfmanstout committed:
Merge branch 'master' into scroll

wolfmanstout committed:
Standardized scrolling flags to use 'wheel' instead of 'scroll' to match Aenea.

wolfmanstout commented:
I'm going to close this and apply to Danesprite/dragonfly instead.
